### PR TITLE
Run a successful cold start on new machine

### PR DESCRIPTION
**Pull request by youngkidwarrior**
I went through the install, build, and dev for both apps in the `/apps` directory, as well as the DB in the `/supabase` directory

[Took notes along the way](https://hackmd.io/@HADkohMGSyy8wLqA3Vdj6w/HkBCnfdST)

This PR includes the changes I needed to make to run the send stack.

This PR is a WIP until I get everything working. Feedback is very much welcome.

# Cold start notes

#### next-app
- Stuck on login. Not sure how to receive code from mobile auth

- linter fails: 
    `yarn next-app lint` 
    
```
Failed to load config "next" to extend from.
Referenced from: /Users/vict0xr/Documents/Send/sendapp/apps/next/.eslintrc.js
```

- NIT: Phone number input form should probably only accept numbers

#### .gitignore
- `generated.ts` is not included in `.gitignore` and there are uncommitted file changes
- `tamagui.css` is not included in `.gitignore` and there are uncommitted file changes

#### 
#### Failed to run native

Platform: iOS ExpoGo

Not running because:
  - `app/features/user/detail-screen' doesn't exist from `[id.tsx]`
  - `'@react-native-community/netinfo'` is not an available package
  - `Error: NEXT_PUBLIC_SUPABASE_URL is not set` from `client.native.ts`
      -  I'm guessing these are set in the EAS development build
  
- transform[stderr]: No native ExponentConstants module found, are you sure the expo-constants's module is linked properly?

##### README
  - `yarn supa` is now `yarn supabase`
  - `yarn setup` command doesn't exist

### CODE REVIEW

File: packages/app/provider/react-query/QueryProvider.native.tsx
[the remark this answers is not in the document]
Comment (youngkidwarrior) [reply packages/app/provider/react-query/QueryProvider.native.tsx:1]: Not broken, `@react-native-community/netinfo` just wasn't available, i.e. not defined in `/packages/app/package.json`

We could also just install `@react-native-community/netinfo` and keep this

[the remark this answers is not in the document]
Comment (0xBigBoss) [reply packages/app/provider/react-query/QueryProvider.native.tsx:1]: yeah that sounds like a decent option, we're pretty far away from adding native support, but would help in the future for sure.

File: packages/wagmi/src/generated.ts
[the remark this answers is not in the document]
Comment (0xBigBoss) [reply packages/wagmi/src/generated.ts:1]: yah in any case, it's probably not a problem, don't worry too much about it.

**Comment by youngkidwarrior** [comment]
@0xBigBoss Collected my notes into the WIP `Contributing.md`

The plan is for me to keep this doc updated following Send's PRs/messaging conversations. That way as new contributors arrive, such as myself, they have some sort of localized overview of the project and its many components

One section I left pretty blank was the `Thinking In Send` Section

This section is meant to explain the mindset you and other contributors are taking while they contribute to various parts of the send stack. I'm pretty new, so I'd like to hear from some team members like you, ethen, leo etc. about it.

I filled it with some basic example headers for now

Thoughts?

The plan in the future is to expand the contributing document for each part of the send stack. Contracts, UI, API, etc.

Once it has grown large enough we can extract these into its own section of the Send docs

**Comment by youngkidwarrior** [comment]
O yea also added some stuff into the root README
```
     ██╗██╗   ██╗███████╗████████╗    ███████╗███████╗███╗   ██╗██████╗     ██╗████████╗
     ██║██║   ██║██╔════╝╚══██╔══╝    ██╔════╝██╔════╝████╗  ██║██╔══██╗    ██║╚══██╔══╝
     ██║██║   ██║███████╗   ██║       ███████╗█████╗  ██╔██╗ ██║██║  ██║    ██║   ██║
██   ██║██║   ██║╚════██║   ██║       ╚════██║██╔══╝  ██║╚██╗██║██║  ██║    ██║   ██║
╚█████╔╝╚██████╔╝███████║   ██║       ███████║███████╗██║ ╚████║██████╔╝    ██║   ██║
 ╚════╝  ╚═════╝ ╚══════╝   ╚═╝       ╚══════╝╚══════╝╚═╝  ╚═══╝╚═════╝     ╚═╝   ╚═╝
```

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+     Don't repeat yourself (DRY)
+     </li>
+     <li>
+     Write a test. Don't click the same button over and over
+     </li>
+     <li>
+     Write once, Run everywhere
+     </li>
+     <li>
+     ?
```
Comment (0xBigBoss) [inline CONTRIBUTING.md:63]: just drop the ? for now.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,193 @@
+# Contribution Guide: WIP
+
+## Preface
+Send Stack uses bleeding-edge web development technology to deliver great DX to our contributors, quick iteration cycles, and clean bug-free code.
```
Comment (0xBigBoss) [inline CONTRIBUTING.md:4]: 🤣 no code is bug-free, users will always find a way to break yo shit. instead clean code is fine or useful would be a better term IMO.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+│   └── <a href="https://github.com/0xsend/sendapp/tree/main/packages/webauthn-authenticator">webauthn-authenticator</a>
+└── <a href="https://github.com/0xsend/sendapp/tree/main/supabase">supabase</a>
+ </code>
+</pre>
+
+
+<details padding="1rem 0">
+<summary style="font-size:20px;font-weight: bold;"><h2 style="display:inline;padding:0 1rem;">Thinking in Send</h2></summary>
+
+Here are some things to keep in mind about thee SEND philosophy when contributing
```
Comment (0xBigBoss) [inline CONTRIBUTING.md:48]: this is a great start. nothing comes to mind immediately.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+then in a new terminal run
+```console
+foundryup
+```
+</details>
+
+
+
+<details style="padding: 1rem 0">
+<summary style="font-size:20px;font-weight:bold;"><h2 style="display:inline;padding:0 1rem;">Your First Build<h2></summary>
```
Comment (0xBigBoss) [inline CONTRIBUTING.md:110]: ty 🙏

**Review by 0xBigBoss — APPROVED** [review]
lgtm

**Pull request merged**